### ISSUE TITLE
Add Kover support for Android Gradle Plugin

### ISSUE THREAD

**Issue by shanshin** [Kotlin/kotlinx-kover#96]
The current implementation of the IntelliJ Coverage Agent allows you to run only Java tests running on the same machine where the Gradle build is running.

To implement coverage measurement on an Android device, it is necessary:
- study how JaCoCo instruments the application on a physical or virtual Android device, how it transmits measurement results
- learn how the Android Gradle Plugin applies JaCoCo to an Android application
- refine the IntelliJ Coverage Agent to run on an Android device and transfer data from it
- implement the correct instrumentation of the Android application in the Kover Gradle plugin

**Comment by Vaibhav1412** [comment]
Are we doing any development around this issue?

**Comment by shanshin** [comment]
@Vaibhav1412, unfortunately, at the moment we have not yet started implementing test instrumentation on Android devices.

After assessing the complexity of the implementation of such functionality, we will be able to more accurately announce the terms.

**Comment by jtomlinson** [comment]
Any updates on how this might be progressing in development or somewhat of an ETA on when we might see this working?

**Comment by shanshin** [comment]
@jtomlinson, this issue is in the plans, but at the moment we are focused on making the API more convenient and adding support for filtering by flavors/build types.
After improving the API, we will start exploring ways to instrument Android devices.

**Comment by pajatopmr** [comment]
As I write this comment, my take on the status of using Kover for Android instrumentation tests is that it is not yet ready but that Jacoco is still usable to some degree. Is that a correct take?

**Comment by shanshin** [comment]
@pajatopmr, yes, at the moment, it is preferable to use JaCoCo for instrumentation tests.

**Comment by merlinstardust** [comment]
Any updates as to when Kover will support Android Instrumentation tests?

If not, how can I pull in the coverage data from JaCoCo and combine it with Kover's so I can get a full report?

**Comment by RobinFarmer** [comment]
Over a year since the last comment I was wondering if there have been any efforts been put into this issue?

**Comment by shanshin** [comment]
Yes, work is underway in this direction

**Comment by stefanoq21** [comment]
Hi @shanshin, do you have any news on that?

**Comment by johngray1965** [comment]
@shanshin I see this is for Milestone "Release 0.9.0", and I see there's a PR for "Release 0.9.0".  Yet there's no update here?

**Comment by shanshin** [comment]
@johngray1965, 
we have reviewed our plans for the development of coverage measurement in Kotlin, and decided to switch to using the JaCoCo agent #720.

This will allow us to integrate with Android Gradle Plugin without having to make changes to AGP itself.

**Comment by Jaehwa-Noh** [comment]
Any updates?
I hope Kover totally able to replace the Jacoco.

**Comment by FrancoisBlavoet** [comment]
Any update or anything the community can do to help unblock this? 
I am not the biggest fan of instrumented tests, but they are sometimes necessary, and not being able to have them contribute to coverage metrics is a pain.
